source/io/module_eggShell.js

```diff
@@ -507,6 +507,56 @@ var assignEggShell;
             Module.stackRestore(stack);
         };
 
+        Module.eggShell.getVariantAttribute = publicAPI.eggShell.getVariantAttribute = function (valueRef, attributeName) {
+            var stack = Module.stackSave();
+
+            var attributeNameStackPointer = Module.coreHelpers.writeJSStringToStack(attributeName);
+            var typeStackPointer = Module.stackAlloc(POINTER_SIZE);
+            var dataStackPointer = Module.stackAlloc(POINTER_SIZE);
+
+            var eggShellResult = Module._EggShell_GetVariantAttribute(Module.eggShell.v_userShell, valueRef.typeRef, valueRef.dataRef, attributeNameStackPointer, typeStackPointer, dataStackPointer);
+            if (eggShellResult !== EGGSHELL_RESULT.SUCCESS && eggShellResult !== EGGSHELL_RESULT.OBJECT_NOT_FOUND_AT_PATH) {
+                throw new Error('Could not get variant attribute for the following reason: ' + eggShellResultEnum[eggShellResult] +
+                    ' (error code: ' + eggShellResult + ')' +
+                    ' (type name: ' + Module.typeHelpers.typeName(valueRef.typeRef) + ')' +
+                    ' (subpath: ' + attributeName + ')');
+            }
+
+            var typeRef, dataRef, subValueRef;
+            if (eggShellResult !== EGGSHELL_RESULT.OBJECT_NOT_FOUND_AT_PATH) {
```

Comment:
This check for `OBJECT_NOT_FOUND_AT_PATH` is the same as above and not necessary, right?

Reply:
Nevermind, it is what the above check is not doing.

Reply:
Nope, there are two cases that make it past the check, success and objectnotfound.

Guess it would be clearer to make the case check for success instead of not objectnotfound

Reply:
Updated the check to be clearer